torch/csrc/jit/register_special_ops.cpp

```diff
@@ -140,7 +140,15 @@ RegisterOperators reg({
     Operator(
         "aten::split(Tensor self, int[] split_sizes, int dim=0) -> Tensor[]",
         [](Stack& stack) {
-          autograd::profiler::RecordFunction record("split_with_sizes");
+          autograd::profiler::RecordFunction record("split_with_sizes", [&stack]() {
+            Stack inputs;
+            const auto& inputs_arr = last(stack, 3);
+            inputs.insert(
+              inputs.end(),
+              inputs_arr.begin(),
+              inputs_arr.end());
+            return inputs;
```

Comment:
This part is repeated so many times, that we should really figure out a way to deduplicate it.

Reply:
Also, this is an overhead for all calls, because every time you call an eager op, you have to instantiate the lambda (which often requires heap allocations) _before you even know if you'll be profiling_.

Reply:
Yeah, I would want to figure out a way to make this more efficient.

For instance, can we make it look like:

```
RECORD_FUNCTION(name, last(stack, 3));
```

Which expands in place to:
```
RecordFunction guard;
switch(profiling_state) {
  case OFF:
       break;
  case NAME_ONLY:
       guard.before();
  case SIZES:
       guard.before(last(stack, 3));));
}
// guard.after() is only called if .before was called.
```

This way we still have a single branch determine record profile information.